guppylang/hugr/hugr.py

```diff
@@ -491,11 +492,13 @@ def add_call(
         self, def_port: OutPortV, args: list[OutPortV], parent: Node | None = None
     ) -> VNode:
         """Adds a `Call` node to the graph."""
-        assert isinstance(def_port.ty, FunctionType)
+        func_ty = is_function(def_port.ty)
+        assert isinstance(func_ty, FunctionType)
```

Comment:
So, `is_function` also accepts tuples. I think this is quite dangerous here since tuples in Hugr are not callable, so you can construct stuff that is invalid Hugr